File: frontend/src/types/plan.ts
export interface ProposedBlock {
  task_id: number | null;
  title: string;
  start_dt: string; // ISO UTC
  end_dt: string;
}

export interface ProposedDayPlan {
  date: string; // YYYY-MM-DD
  blocks: ProposedBlock[];
  unplaced_task_ids: number[];
  fully_booked: boolean;
}

export interface ScheduledBlock {
  id: number;
  task_id: number | null;
  title: string;
  start_dt: string;
  end_dt: string;
  date_key: string;
  approved_at: string;
  completed: boolean;
  conflict_with: string | null;
}

File: backend/app/schemas/plan.py
from datetime import date, datetime

from pydantic import BaseModel


class ProposedBlock(BaseModel):
    task_id: int | None = None
    title: str
    start_dt: datetime
    end_dt: datetime


class ProposedDayPlan(BaseModel):
    date: date
    blocks: list[ProposedBlock]
    unplaced_task_ids: list[int]
    fully_booked: bool


class ApproveRequest(BaseModel):
    date: date
    blocks: list[ProposedBlock]


class ScheduledBlockRead(BaseModel):
    id: int
    task_id: int | None
    title: str
    start_dt: datetime
    end_dt: datetime
    date_key: str
    approved_at: datetime
    completed: bool
    conflict_with: str | None = None
    model_config = {"from_attributes": True}


class ScheduledBlockUpdate(BaseModel):
    completed: bool
